subprojects/android-test/snackbar-rule/src/main/kotlin/com/avito/android/test/app/second/SnackbarProxyRule.kt

```diff
@@ -0,0 +1,62 @@
+package com.avito.android.test.app.second
+
+import android.annotation.SuppressLint
+import android.widget.TextView
+import com.avito.android.rule.SimpleRule
+import com.avito.android.snackbar.proxy.SnackbarProxy
+import com.avito.android.snackbar.proxy.SnackbarProxyHolder
+import com.avito.android.test.waitFor
+import com.google.android.material.snackbar.Snackbar
+import org.junit.Assert
+import com.google.android.material.R as google_R
+
+class SnackbarProxyRule : SimpleRule() {
+
+    private val proxy = Proxy()
+
+    override fun before() {
+        SnackbarProxyHolder.proxy = proxy
+    }
+
+    fun isShown(text: String) {
```

Comment:
Consider adding an interface `SnackbarChecks` and a field `checks` to follow framework style

Reply:
NP